src/core/reopt.jl

```diff
@@ -652,14 +741,14 @@ function add_variables!(m::JuMP.AbstractModel, p::REoptInputs)
 		# TODO: currently defining more decision variables than necessary b/c using rectangular arrays, could use dicts of decision variables instead
         @variables m begin # if there is more than one specified outage, there can be more othan one outage start time
 			dvUnservedLoad[S, tZeros, outage_time_steps] >= 0 # unserved load not met by system
-			dvMGProductionToStorage[p.techs.elec, S, tZeros, outage_time_steps] >= 0 # Electricity going to the storage system during each time_step
-			dvMGDischargeFromStorage[S, tZeros, outage_time_steps] >= 0 # Electricity coming from the storage system during each time_step
-			dvMGRatedProduction[p.techs.elec, S, tZeros, outage_time_steps]  # MG Rated Production at every time_step.  Multiply by production_factor to get actual energy
-			dvMGStoredEnergy[S, tZeros, 0:max_outage_duration] >= 0 # State of charge of the MG storage system
+			dvMGProductionToStorage[union(p.s.storage.types.elec, p.s.storage.types.hydrogen), union(p.techs.elec, p.techs.electrolyzer, p.techs.compressor), S, tZeros, outage_time_steps] >= 0 # Electricity going to the storage system during each time_step
+			dvMGDischargeFromStorage[union(p.s.storage.types.elec, p.s.storage.types.hydrogen), S, tZeros, outage_time_steps] >= 0 # Electricity coming from the storage system during each time_step
```

Comment:
would this just be p.s.storage.types.all?

Reply:
> would this just be p.s.storage.types.all?

My read of this is that we're excluding thermal techs that may be in the broader optimization but not the electrical part of the microgrid.  That's consistent with the existing code, but I guess that with the addition of hydrogen we preclude thermal storage or a heating tech from serving something like a steam turbine - is that what we want?